server/context.go

```diff
@@ -249,6 +263,42 @@ func (ctx *Context) InitNode() error {
 	return nil
 }
 
+// parseDurationEnv parses a time.Duration from an environment variable. This
+// function assumes that the default value is already present in duration.
+func parseDurationEnv(env, internalName string, duration *time.Duration) {
+	if valueString := os.Getenv(env); len(valueString) != 0 {
+		if value, err := time.ParseDuration(valueString); err != nil {
+			log.Errorf("could not parse environment variable %s=%s, setting to default of %s, error: %s",
+				env, valueString, duration, err)
+		} else {
+			*duration = value
+			log.Infof("\"%s\" set to %s based on %s environment variable", internalName, *duration, env)
+		}
+	}
+}
+
+// readEnvironmentVariables populates all context values that are environment
+// variable based. Note that this only happens when initializing a node and not
+// when NewContext is called.
+func (ctx *Context) readEnvironmentVariables() {
+	// cockroach-linearizable
+	if linearizableString := os.Getenv("cockroach-linearizable"); len(linearizableString) != 0 {
+		if linearizable, err := strconv.ParseBool(linearizableString); err != nil {
+			log.Errorf("could not parse environment variable cockroach-linearizable=%s, setting to default of %t, error: %s",
+				linearizableString, ctx.Linearizable, err)
+		} else {
+			ctx.Linearizable = linearizable
+			log.Infof("\"linearizable\" set to %t based on cockroach-linearizable environment variable", ctx.Linearizable)
+		}
+	}
+
+	parseDurationEnv("cockroach-max-offset", "max offset", &ctx.MaxOffset)
```

Comment:
Environment variables are almost always all caps and separated with underscores. While this isn't required, people will look at us funny if we use lowercase env variables.